### ISSUE TITLE
Expose followers-only protocol internals to the API (no protocol/UI changes)

### ISSUE THREAD

**Issue by SoniEx2** [tootsuite/mastodon#13586]
### Pitch

Followers-only posts are simply posts addressed to "followers of <self>", but there's no protocol-level reason why it can't be addressed to "followers of <other>" - masto just doesn't expose it. I think masto should just expose it.

### Motivation

Replies to followers-only posts should have flexible addressing. They already do - you can make it public, unlisted, "followers of <self>", or "only mentioned users". This would let advanced users use "followers of <OP>" instead, by using a custom client.

**Comment by trwnh** [comment]
#11183

**Comment by SoniEx2** [comment]
well, this one *is* a little broader in scope, tbh. (see: mailing lists)

**Comment by SoniEx2** [comment]
I should expand on the mailing list idea: basically you could run AP software that has mailing lists.

a mailing list is an actor. they can boost posts. so you @ it + set the privacy to "followers of <it>". it'll then boost the post to original audience (aka "followers of <it>"), and you can subscribe to that mailing list by following the actor.

**Comment by trwnh** [comment]
>a mailing list is an actor. they can boost posts. so you @ it + set the privacy to "followers of <it>". it'll then boost the post to original audience (aka "followers of <it>"), and you can subscribe to that mailing list by following the actor.

You're describing Group actors and inbox forwarding. You don't have to know the followers of the Group -- the Group knows its own followers, so the way it works is to take any mentions it receives and send the Announce to its members.

**Comment by SoniEx2** [comment]
does it matter whether it's a Group actor or not?

**Comment by trwnh** [comment]
No, it doesn't matter. In general, types don't really matter. What matters is the software running on the other end.

I think Friendica's Groups work similar to a mailing list, though...

**Comment by SoniEx2** [comment]
ah, okay.

well, in any case mastodon's inability to address a mailing list's subscribers kinda makes one unable to use those from masto...

**Comment by ThibG** [comment]
If I understand your request correctly, you want to be able to write toots visible to *someone else's followers*. This actually brings a few issues. Indeed, in the general case, Mastodon does not know about that other person's followers and thus cannot deliver the message itself. A solution would be to make the other person's instance forward the toot, but this would require LDSigning the toot, which we only do for public/unlisted toots and when authorized fetch is disabled. Indeed LDSigning toots, private or otherwise:
- makes it difficult to deny you wrote it, even in the future
- allows people to pass it around with proof you wrote it, including to blocked users (knowingly or not, there could be people you block among the person's followers, for instance)

**Comment by SoniEx2** [comment]
maybe this is where we could benefit from that OCapPub stuff.

have the server send a bearer instead of LDSigning.

**Comment by SoniEx2** [comment]
I feel like I came off unnecessarily aggressive on this issue, sorry about that. it won't happen again.
